submarine-server/server-api/src/main/java/org/apache/submarine/server/api/spec/JobSpec.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.server.api.spec;
+
+import java.util.Map;
+
+/**
+ * The submarine job spec for submarine job server. It consist of name, JobLibrarySpec,
+ * JobSubmitterSpec, the tasks.
+ */
+public class JobSpec {
+  /**
+   * The user-specified job name. Such as: mnist
+   */
+  private String name;
+
+  /**
+   * The user-specified ML framework spec.
+   */
+  private JobLibrarySpec librarySpec;
+
+  /**
+   * The user-specified submitter spec.
+   */
+  private JobSubmitterSpec submitterSpec;
+
+  /**
+   * The tasks of the job.
+   * Such as:
+   *   TensorFlow: Chief, Ps, Worker, Evaluator
+   *   PyTorch: Master, Worker
+   */
+  private Map<String, JobTaskSpec> taskSpecs;
```

Comment:
I'm confusing here. What's the key and value of the taskSpecs? What's the key difference between JobTaskSpec and the deleted "component.java"?

Reply:
@tangzhankun  The map's key is `Master` or `Worker` etc, I will add the more detailed comment for it.